source/extensions/common/wasm/context.cc

```diff
@@ -1305,6 +1305,32 @@ WasmResult Context::continueStream(StreamType stream_type) {
   return WasmResult::Ok;
 }
 
+WasmResult Context::closeStream(WasmStreamType stream_type) {
+  switch (stream_type) {
+  case WasmStreamType::Request:
+    if (decoder_callbacks_) {
+      decoder_callbacks_->resetStream();
+    }
+    return WasmResult::Ok;
+  case WasmStreamType::Response:
+    if (encoder_callbacks_) {
+      encoder_callbacks_->resetStream();
+    }
+    return WasmResult::Ok;
+  case WasmStreamType::Downstream:
+    if (network_read_filter_callbacks_) {
+      network_read_filter_callbacks_->connection().close(
+          Envoy::Network::ConnectionCloseType::FlushWrite);
```

Comment:
Why always FlushWrite?

Reply:
@PiotrSikora and I had a long talk about this.  We certainly do not want to wait, so it was either WriteFlush or NoFlush.  We decided on WriteFlush although to be honest I could go either way.